e-commerce-frontend/src/app/_services/user.service.ts

```diff
@@ -0,0 +1,65 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { UserAuthService } from './user-auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+
+  PATH_OF_API = 'https://localhost:8080';
+
+  requestHeaders = new HttpHeaders(
+    { 'No-Auth': 'True' }
+  );
+  constructor(private httpClient: HttpClient, private userAuthService: UserAuthService) {
+  }
+
+  //send a login request with user credentials
+  public login(loginData: any) {
+    return this.httpClient.post(this.PATH_OF_API + "/authenticate", loginData, { headers: this.requestHeaders });
+  }
+
+  //get data for users with the 'SELLER' role
+  public forSeller() {
+    return this.httpClient.get(this.PATH_OF_API + "/forSeller", { responseType: 'text' });
+  }
+
+  //get data for users with the 'BUYER' role
+  public forBuyer() {
+    return this.httpClient.get(this.PATH_OF_API + "/forBuyer", { responseType: 'text' });
+  }
+
+  //get data for users with 'ADMIN' role
+  public forAdmin() {
+    return this.httpClient.get(this.PATH_OF_API + "/forAdmin", { responseType: 'text' });
+  }
```

Comment:
I think those methods should have a more descriptive name like: "getDataForSeller".